packages/embark-api-client/src/index.js

```diff
@@ -0,0 +1,281 @@
+import axios from "axios";
+import keccak from "keccakjs";
+
+function hash(cnonce, token, type, path) {
+  let hash = new keccak();
+  hash.update(cnonce.toString());
+  hash.update(token);
+  hash.update(type.toUpperCase());
+  hash.update(`/embark-api${path}`);
+  return hash.digest('hex');
+}
+
+function request(type, path, params = {}) {
+  const cnonce = Date.now() + Math.random();
+
+  // Extract credentials out of the params and delete so the token doesn't get sent
+  // as cleartext.
+  let credentials = params.credentials;
+  delete params.credentials;
+
+  let requestHash = hash(cnonce, credentials.token, type, path, params);
+
+  const endpoint = `http://${credentials.host}/embark-api${path}`;
+  const req = {
+    method: type,
+    url: endpoint,
+    headers: {
+      'X-Embark-Request-Hash': requestHash,
+      'X-Embark-Cnonce': cnonce
+    },
+    ...(type === 'post' ? { data: params } : {}),
+    ...(['get', 'delete'].includes(type) ? { params: params.params } : {})
+  };
+
+  return axios(req)
+    .then((response) => {
+      return (response.data && response.data.error) ? {error: response.data.error} : {response, error: null};
+    }).catch((error) => {
+      return {response: null, error: error.message || 'Something bad happened'};
+    });
+}
+
+function get() {
+  console.dir("== get");
+  console.dir(...arguments);
+  return request('get', ...arguments);
+}
+
+function post() {
+  console.dir("== post");
+  console.dir(...arguments);
+  return request('post', ...arguments);
+}
+
+function destroy() {
+  console.dir("== destroy");
+  console.dir(...arguments);
+  return request('delete', ...arguments);
+}
+
+function websocket(credentials, path) {
+  const cnonce = Date.now() + Math.random();
+  const requestHash = hash(cnonce, credentials.token, 'ws', '/', {});
+
+  return new WebSocket(`ws://${credentials.host}/embark-api${path}`, [`${cnonce}|${requestHash}`]);
+}
+
+class EmbarkAPI {
```

Comment:
Wonder if we could call this `RestAPI` or something like that.

Also, should the class match the name of the package?

Reply:
Agreed re: matching. How about EmbarkWebApi and embark-web-api?